torch/_dynamo/variables/builder.py

```diff
@@ -920,3 +925,36 @@ def _clone_input(value):
             "torch.* op returned non-Tensor "
             + f"{typestr(example_value)} {proxy.node.op} {proxy.node.target}"
         )
+
+
+# Tracks the sources of all fake tensors we wrap in Dynamo.
+# Used by shape guard computation.
+@dataclasses.dataclass
+class FakeSource:
+    fake: Union[FakeTensor, SymInt]
+    sname: str
+
+
+def wrap_to_fake_tensor_and_record(
```

Comment:
If this were my PR, you would get me for unnecessary code motion :P

Reply:
:P